File: 125 Valid Palindrome/Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int l = 0, r = s.length() - 1;

        while (l < r) {
            while (l < r && !isAlpha(s.charAt(l))) {
                l++;
            }

            while (r > l && !isAlpha(s.charAt(r))) {
                r--;
            }

            if (Character.toLowerCase(s.charAt(l)) != Character.toLowerCase(s.charAt(r))) {
                return false;
            }

            l++;
            r--;
        }

        return true;
    }

    private boolean isAlpha(char letter) {
        return (letter >= 'A' && letter <= 'Z') || (letter >= 'a' && letter <= 'z') || (letter >= '0' && letter <= '9');
    }
}

File: 125 Valid Palindrome/Valid Palindrome.cpp
class Solution
{
public:
    bool isPalindrome(string s)
    {
        int l = 0, r = s.length() - 1;

        while (l < r)
        {
            while (l < r && !isAlpha(s[l]))
            {
                l++;
            }

            while (r > l && !isAlpha(s[r]))
            {
                r--;
            }

            if (tolower(s[l]) != tolower(s[r]))
            {
                return false;
            }

            l++;
            r--;
        }

        return true;
    }

private:
    bool isAlpha(char letter)
    {
        return (letter >= 'A' && letter <= 'Z') || (letter >= 'a' && letter <= 'z') || (letter >= '0' && letter <= '9');
    }
};
